File: 2285.cpp
struct Solution {
    long long maximumImportance(int n, vector<vector<int>>& roads) {
        vector<int> deg(n, 0);
        for (auto &r : roads){
            ++deg[r[0]];
            ++deg[r[1]];
        }
        sort(deg.begin(), deg.end());
        long long res = 0;
        for (long long i = 0; i < n; ++i){
            res += (i + 1) * deg[i];
        }
        return res;
    }
};

File: sol/2285/2285.py
class Solution:
    def maximumImportance(self, n: int, roads: List[List[int]]) -> int:
        deg = [0] * n
        for a, b in roads:
            deg[a] += 1
            deg[b] += 1
        deg.sort()
        res = 0
        for i, x in enumerate(deg):
            res += x * (i + 1)
        return res
